api/views/assets.py

```diff
@@ -576,6 +576,15 @@ class StateTransitionViewset(ModelViewSet):
     def perform_update(self, serializer):
 
         try:
+            report_sate = self.request.data.get("incident_report_state", None)
+            asset_state = self.request.data.get("asset_state_from_report", None)
+            if (
+                report_sate == "external assessment"
+                and asset_state == "requires external assessment"
+            ):
+                raise serializers.ValidationError(
```

Comment:
I don't think it's nice raising exceptions in a try block which is meant to catch exceptions. You could just do your data integrity check and raise this exception outside the try block.